monitor/internal/docker/monitor.go

```diff
@@ -221,17 +221,34 @@ func (d *DockerMonitor) eventProcessors(ctx context.Context) {
 // that we will miss events because the processor is delayed
 func (d *DockerMonitor) eventListener(ctx context.Context, listenerReady chan struct{}) {
 
+	// Once the buffered event channel was returned by Docker we return the ready status.
+	listenerReady <- struct{}{}
+
+	for {
+		if d.dockerClient == nil {
+			zap.L().Debug("Trying to setup docker daemon")
+			if err := d.setupDockerDaemon(); err != nil {
+				d.dockerClient = nil
+				continue
+			}
+			// We are here means the docker daemon restarted. we need to resync
+			if err := d.Resync(ctx); err != nil {
+				zap.L().Error("Unable to resync containers after reconnecting to docker daemon", zap.Error(err))
+			}
+		}
+		d.listener(ctx)
+
+	}
+}
+
```

Comment:
I think it needs a lock. There are two go-routines fighting for the d.dockerClient

Reply:
Added a lock around accessing dockerClient